### ISSUE TITLE
False negative on NoReturn async def

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#2880]
```python
from typing import NoReturn

async def noret() -> NoReturn:
    raise NotImplementedError

async def pacarana(cond) -> int:
    if cond:
        return 3
    else:
        x = noret()
        print(x)
```
This doesn't produce any errors in latest Pylance. If I remove the `else` block, it correctly says `Function with declared type of "int" must return value`.

**Comment by erictraut** [comment]
Thanks for the bug report. I've updated the logic to require that a call to an async function that returns `NoReturn` needs to be awaited to be considered a `NoReturn` call from the perspective of the code flow engine.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.211, which I just published. It will also be included in the next release of pylance.
